core/templates/pages/exploration-player-page/learner-experience/conversation-skin.component.spec.ts

```diff
@@ -1298,6 +1298,24 @@ describe('Conversation skin component', () => {
     expect(componentInstance.mostRecentlyReachedCheckpoint).toBe('Mid');
   }));
 
+  it('should display the exploration after the the progress reminder modal' +
+   'has loaded', () => {
+    componentInstance.CHECKPOINTS_FEATURE_IS_ENABLED = true;
+    spyOn(contextService, 'isInExplorationEditorPage').and.returnValue(false);
+    spyOn(contextService, 'isInExplorationPlayerPage').and.returnValue(true);
+    spyOn(urlService, 'getCollectionIdFromExplorationUrl').and.returnValue(
+      null);
+    spyOn(urlService, 'getPidFromUrl').and.returnValue(null);
+    spyOn(explorationEngineService, 'getExplorationId').and.returnValue(
+      'expl_1');
+    spyOn(explorationEngineService, 'isInPreviewMode').and.returnValue(false);
+    spyOn(urlService, 'isIframed').and.returnValue(false);
+
+    componentInstance.ngOnInit();
+    explorationPlayerStateService.onShowProgressModal.emit();
+    expect(componentInstance.hasFullyLoaded).toBe(true);
```

Comment:
Should there be a pre-check for this?

Reply:
Do you mean to put a check between lines 1314 and 1315 if hasFullyLoaded is false and then true after it?

Reply:
Yes, that's right. Also, in the future, please feel free to make the changes and then respond to the comments. It is faster that way, if there is some error in the way a comment is addressed, we can always revert / fix that.

Reply:
Done